services/spar/src/Spar/App.hs

```diff
@@ -199,19 +200,39 @@ insertUser uref uid = wrapMonadClient $ Data.insertSAMLUser uref uid
 -- the team with valid SAML credentials.
 --
 -- FUTUREWORK: Remove and reinstatate getUser, in AuthID refactoring PR.  (in https://github.com/wireapp/wire-server/pull/1410, undo https://github.com/wireapp/wire-server/pull/1418)
-getUserByUref :: SAML.UserRef -> Spar (Maybe UserId)
-getUserByUref uref = do
+getUserIdByUref :: Maybe TeamId -> SAML.UserRef -> Spar (GetUserResult UserId)
+getUserIdByUref mbteam uref = userId <$$> getUserByUref mbteam uref
+
+getUserByUref :: Maybe TeamId -> SAML.UserRef -> Spar (GetUserResult User)
+getUserByUref mbteam uref = do
   muid <- wrapMonadClient $ Data.getSAMLUser uref
   case muid of
-    Nothing -> pure Nothing
+    Nothing -> pure GetUserNotFound
     Just uid -> do
       let withpending = Intra.WithPendingInvitations -- see haddocks above
-      itis <- isJust <$> Intra.getBrigUserTeam withpending uid
-      pure $ if itis then Just uid else Nothing
+      Intra.getBrigUser withpending uid >>= \case
+        Nothing -> pure GetUserNotFound
+        Just user
+          | isNothing (userTeam user) -> pure GetUserNoTeam
+          | mbteam /= userTeam user -> pure GetUserWrongTeam
+          | otherwise -> pure $ GetUserFound user
+
+data GetUserResult usr
+  = GetUserFound usr
+  | GetUserNotFound
+  | GetUserNoTeam
+  | GetUserWrongTeam
+  deriving (Eq, Show)
```

Comment:
I think you can just `deriving Functor`.

Reply:
Thanks, will fix in an upcoming PR.  (Also for `GetIdPResult` in `Spar.Data`.)